source/screenExplorer.py

```diff
@@ -1,8 +1,59 @@
+#screenExplorer.py
+#A part of NonVisual Desktop Access (NVDA)
+#Copyright (C) 2012-2014 NVDA Contributors
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+
+"""Module which implements various utilities for screen exploration.
+This includes touchscreen movement as well as handling coordinate announcement for mouse and objects.
+"""
+
 import api
 import tones
 import controlTypes
 import textInfos
 import speech
+import screenBitmap
+import config
+import mouseHandler
+
+def playLocationCoordinates(x, y, screenWidth, screenHeight, detectBrightness=True,blurFactor=0):
+	"""Plays a tone indicating the XY-coordinate for the given location. This works for both mouse movement and during touch hover gesture.  
+	@param obj: the coordinate for the location and the screen resolution (typically desktopObject.location)
+	@type obj: int
+	"""
+	if x > screenWidth or y > screenHeight:
+		return
+	minPitch=config.conf['mouse']['audioCoordinates_minPitch']
+	maxPitch=config.conf['mouse']['audioCoordinates_maxPitch']
+	curPitch=minPitch+((maxPitch-minPitch)*((screenHeight-y)/float(screenHeight)))
+	if detectBrightness:
+		startX=min(max(x-blurFactor,0),screenWidth)
+		width=min((x+blurFactor+1)-startX,screenWidth)
+		startY=min(max(y-blurFactor,0),screenHeight)
+		height=min((y+blurFactor+1)-startY,screenHeight)
+		grey=screenBitmap.rgbPixelBrightness(mouseHandler.scrBmpObj.captureImage(startX,startY,width,height)[0][0])
+		brightness=grey/255.0
+		minBrightness=config.conf['mouse']['audioCoordinates_minVolume']
+		maxBrightness=config.conf['mouse']['audioCoordinates_maxVolume']
+		brightness=(brightness*(maxBrightness-minBrightness))+minBrightness
+	else:
+		brightness=config.conf['mouse']['audioCoordinates_maxVolume']
+	leftVolume=int((85*((screenWidth-float(x))/screenWidth))*brightness)
+	rightVolume=int((85*(float(x)/screenWidth))*brightness)
+	tones.beep(curPitch,40,left=leftVolume,right=rightVolume)
+
+def playObjectCoordinates(obj):
+	"""Plays the audio coordinate for the object.  
+	@param obj: the object for which the coordinate announced should be played
+	@type obj: NVDAObjects.NvDAObject
+	"""
+	l,t,w,h=obj.location
+	x = l+(w/2)
+	y = t+(h/2)
+	screenWidth, screenHeight = api.getDesktopObject().location[2], api.getDesktopObject().location[3]

```

Comment:
Rather than calling getDesktopObject twice,  just slice the location:
screenWidth,screenHeight=api.getDesktopObject().location[2:]